lib/fetch/request.js

```diff
@@ -420,10 +420,9 @@ class Request {
       // 4. If headers is a Headers object, then for each header in its header
       // list, append header’s name/header’s value to this’s headers.
       if (headers instanceof Headers) {
-        // TODO (fix): Why doesn't this work?
-        // for (const [key, val] of headers[kHeadersList]) {
-        //   this[kHeaders].append(key, val)
-        // }
+        for (const [key, val] of headers) {
+          this[kHeaders].append(key, val)
+        }
```

Comment:
I think the `this[kState].headersList = new HeadersList([` below should be removed then.

Reply:
Done in 2569544a67c30d1e74177c51b2105a4181f09c74 👍 

I also found a way to remove the `kInit` overload in the constructor, so I'll probably make a new PR for that at some point.

Reply:
So removing this entirely caused node-fetch tests to fail, however just using the newly constructed headers' header list did work.